bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RackawareEnsemblePlacementPolicyImpl.java

```diff
@@ -456,7 +456,10 @@ public void uninitalize() {
             excludeBookies.addAll(currentEnsemble);
             BookieNode bn = knownBookies.get(bookieToReplace);
             if (null == bn) {
-                bn = createBookieNode(bookieToReplace);
+                bn = historyBookies.get(bookieToReplace);
+                if (null == bn) {
+                    bn = createBookieNode(bookieToReplace);
```

Comment:
probably it is better to try to use createBookieNode and fallback to history if it fails.
otherwise you won't be able to follow a change in the network location for the bookie.

Reply:
Thanks for your quickly feedback. OK, it's better to fallback to history when resolve bookieId failed, it will update the code.